### ISSUE TITLE
The reason why the author's results cannot be reproduced

### ISSUE THREAD

**Issue by ZhengChen6** [ge-xing/Diff-UNet#32]
作者您好！
我猜测在issue种很多人无法复现您的结果是因为您在您的代码种用os.listdir获得目录然后再划分训练集和测试集，一旦更换主机，os.listdir得到的文件的顺序是不一样的，所以我们最终划分的数据集和您的数据集是不一样的。我分别尝试在两台主机上打印os.listdir获得的目录顺序，发现其不一致。
![image](https://github.com/ge-xing/Diff-UNet/assets/103295650/780ec097-aa26-4cd0-9030-d68ac56f9e66)

主机一：
![image](https://github.com/ge-xing/Diff-UNet/assets/103295650/8e478382-f3c4-44c4-a7ec-61a28c5b4329)

主机二：
![image](https://github.com/ge-xing/Diff-UNet/assets/103295650/594c09a9-8b51-4b17-ad13-c35fcd34bd32)

我建议您对os.listdir后的all_dirs进行排序后再划分数据集！

I guess that many people in the issue cannot reproduce your results because you use os.listdir in your code to obtain the dirs and then divide the training set and test set. Once the host is changed, the order of the files obtained by os.listdir is not the same. The same, so the data set we finally divided is different from your data set. I tried to print the directory order obtained by os.listdir on the two hosts and found that it was inconsistent. @ge-xing

**Comment by ge-xing** [comment]
Thank you~

**Comment by zz973** [comment]
作者您好如何编写代码保存模型的预测分割图，我看你是有写到图片保存函数的但是你没有调用

**Comment by ge-xing** [comment]
validation_step 里面 得到预测结果以后，调用sitk的函数保存为nii gz文件就行
